.nvidia-ci.yml

```diff
@@ -419,3 +419,241 @@ release:ngc-rhel9.3:
+    - chmod a+x bin/regctl
+    - export PATH=$(pwd)/bin:${PATH}
+    - |
+      if [ -z "${NGCCLI_VERSION}" ]; then
+        NGC_VERSION_URL="https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions"
+        # Extract the latest version from the JSON data using jq
+        export NGCCLI_VERSION=$(curl -s $NGC_VERSION_URL | jq -r '.recipe.latestVersionIdStr')
+      fi
+      echo "NGCCLI_VERSION ${NGCCLI_VERSION}"
+    - curl -sSLo ngccli_linux.zip https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions/${NGCCLI_VERSION}/files/ngccli_linux.zip
+    - unzip ngccli_linux.zip
+    - chmod u+x ngc-cli/ngc
+
+# .sign forms the base of the deployment jobs which signs images in the CI registry.
+# This is extended with the image name and version to be deployed.
+.sign:ngc:
+  image: ubuntu:latest
+  timeout: 30 minutes
+  when: always
+  stage: sign
+  rules:
+    - if: $CI_COMMIT_TAG
+  variables:
+    OUT_REGISTRY_USER: "${CI_REGISTRY_USER}"
+    OUT_REGISTRY_TOKEN: "${CI_REGISTRY_PASSWORD}"
+    OUT_REGISTRY: "${CI_REGISTRY}"
+    NGC_CLI_API_KEY: "${NGC_REGISTRY_TOKEN}"
+    IMAGE_NAME: "${NGC_REGISTRY_IMAGE}"
+    IMAGE_TAG: "${CI_COMMIT_TAG}"
+  before_script:
+    - !reference [.ngccli-setup, before_script]
+    - 'if [[ "${PRECOMPILED}" != "true" ]]; then echo DRIVER_VERSION: ${DRIVER_VERSION} ; [[ -n "${DRIVER_VERSION}" ]] || exit 1; fi'
+    - regctl registry login "${OUT_REGISTRY}" -u "${OUT_REGISTRY_USER}" -p "${OUT_REGISTRY_TOKEN}"
+    - |
+      if [[ "${PRECOMPILED}" == "true" ]]; then
+        ./scripts/precompiled.sh version;
+        source kernel_version.txt
+        export IMAGE_TAG=${DRIVER_BRANCH}-${KERNEL_VERSION}-${DIST##*_}
+      else
+          export IMAGE_TAG=${DRIVER_VERSION}-${DIST##*_}
+      fi
+    # We ensure that the IMAGE_NAME and IMAGE_TAG is set
+    - 'echo Image Name: ${IMAGE_NAME} && [[ -n "${IMAGE_NAME}" ]] || exit 1'
+    - 'echo Image Tag: ${IMAGE_TAG} && [[ -n "${IMAGE_TAG}" ]] || exit 1'
+  script:
+    - 'echo "Signing the image ${IMAGE_NAME}:${IMAGE_TAG}"'
+    - ngc-cli/ngc registry image publish --source ${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_NAME}:${IMAGE_TAG} --public --discoverable --allow-guest --sign --org nvidia
+
+sign:ngc-precompiled-ubuntu22.04:
+  extends:
+    - .driver-versions-precompiled-ubuntu22.04
+    - .dist-ubuntu22.04
+    - .release-generic
+    - .release:ngc-variables
+    - .sign:ngc
+  variables:
+    DIST: signed_ubuntu22.04
+    BASE_TARGET: jammy
+    PRECOMPILED: "true"
+  needs:
+    - release:ngc-precompiled-ubuntu22.04
+
+sign:ngc-ubuntu22.04:
+  extends:
+    - .release-ubuntu22.04
+    - .dist-ubuntu22.04
+    - .driver-versions-ubuntu22.04
+    - .sign:ngc
+  needs:
+    - release:ngc-ubuntu22.04
+
+sign:ngc-ubuntu20.04:
+  extends:
+    - .release
+    - .dist-ubuntu20.04
+    - .sign:ngc
+  needs:
+    - release:ngc-ubuntu20.04
+
+sign:ngc-rhcos4.9:
```

Comment:
Is it possible to use a matrix so that we have one `sign:ngc-rhcos` job and we specify a matrix of `RHCOS_VERSIONS` (e.g. 4.9, 4.10, ...)? The same suggestion applies for the rhel jobs.

Reply:
This may require us to also parallelize the `release:ngc-rhcos` jobs using similar logic. If required we can improve this in a follow-up.

Reply:
added : ubuntu 22.04, 20.04 , rhcos and rhel in parallel matrix .
@cdesiniotis check if thel required .
 - SIGN_JOB_NAME: ["rhel" ]
      VERSION: ["8.6", "8.7", "8.8", "8.9","8.10"]
      DRIVER_VERSION: ["470.256.02", "535.183.01", "550.90.07"]